src/extractLoader.js

```diff
@@ -28,7 +28,9 @@ const rndPlaceholder = "__EXTRACT_LOADER_PLACEHOLDER__" + rndNumber() + rndNumbe
 function extractLoader(content) {
     const callback = this.async();
     const options = getOptions(this) || {};
-    const publicPath = options.publicPath === undefined ? this.options.output.publicPath : options.publicPath;
+    const publicPath = options.publicPath ||
+        (this.options && this.options.output && this.options.output.publicPath) ||
+        (this._compilation && this._compilation.outputOptions && this._compilation.outputOptions.publicPath);
```

Comment:
1. I think we should check for `"publicPath" in options` here because an empty string as `publicPath` is a valid option
2. The check for `this.options` is necessary with webpack 4, but could you add a `TODO` that this should be removed with the next major release
3. I guess you've added the `this._compilation` part so that we can still derive the `publicPath`. We should also remove that with the next major release since [`this._compilation` is likely to be removed in the future](https://webpack.js.org/api/loaders/#deprecated-context-properties).

Reply:
I've created an issue for that: https://github.com/peerigon/extract-loader/issues/35

Reply:
1. You're right. I'll change that.
2. and 3. #35 is enough, isn't it?